File: searxng-client/quickstart.js
#!/usr/bin/env node

/**
 * Quick Start Example for SearXNG API
 * 
 * This script demonstrates the simplest way to get started with the SearXNG API.
 */

const { SearXNGClient } = require('./searxng-client');

async function main() {
    // Initialize the client
    const client = new SearXNGClient();
    
    console.log('🚀 SearXNG API Quick Start');
    console.log('='.repeat(60));
    
    // Example 1: Simple search
    console.log('\n📝 Example 1: Simple Search');
    console.log('-'.repeat(60));
    try {
        const results = await client.searchGeneral('python requests library');
        console.log(`Found ${results.results.length} results`);
        if (results.results.length > 0) {
            console.log(`First result: ${results.results[0].title}`);
            console.log(`URL: ${results.results[0].url}`);
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
    
    // Example 2: Code search
    console.log('\n💻 Example 2: Code Search');
    console.log('-'.repeat(60));
    try {
        const results = await client.searchCode('how to use async await in javascript');
        console.log(`Found ${results.results.length} code-related results`);
        if (results.results.length > 0) {
            console.log(`First result: ${results.results[0].title}`);
            console.log(`URL: ${results.results[0].url}`);
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
    
    // Example 3: Academic search
    console.log('\n🎓 Example 3: Academic Search');
    console.log('-'.repeat(60));
    try {
        const results = await client.searchAcademic('transformer neural networks');
        console.log(`Found ${results.results.length} academic papers`);
        if (results.results.length > 0) {
            console.log(`First paper: ${results.results[0].title}`);
            console.log(`URL: ${results.results[0].url}`);
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
    
    // Example 4: Custom search with specific engines
    console.log('\n🎯 Example 4: Custom Search');
    console.log('-'.repeat(60));
    try {
        const results = await client.search('machine learning tutorial', {
            engines: ['github', 'brave']
        });
        console.log(`Found ${results.results.length} results from GitHub and Brave`);
        if (results.results.length > 0) {
            console.log(`First result: ${results.results[0].title}`);
            console.log(`Engine: ${results.results[0].engine}`);
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
    
    console.log('\n✅ Quick start complete!');
    console.log('\n💡 Next steps:');
    console.log('   - Import the client in your code: const { SearXNGClient } = require(\'./searxng-client\');');
    console.log('   - Check the README.md for detailed API documentation');
    console.log('   - Explore the available search engines and options');
    console.log('\n🌐 API Endpoint: https://searxng-production-b099.up.railway.app');
}

main().catch(console.error);

File: searxng-client/quickstart.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Quick Start Example for SearXNG API

This script demonstrates the simplest way to get started with the SearXNG API.
"""

import sys
import io

# Set stdout to handle UTF-8 on Windows
if sys.platform == "win32":
    sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding="utf-8")

from searxng_client import SearXNGClient


def main():
    # Initialize the client
    client = SearXNGClient()

    print("🚀 SearXNG API Quick Start")
    print("=" * 60)

    # Example 1: Simple search
    print("\n📝 Example 1: Simple Search")
    print("-" * 60)
    results = client.search_general("python requests library")
    print(f"Found {len(results['results'])} results for 'python requests library'")
    if results["results"]:
        print(f"First result: {results['results'][0]['title']}")
        print(f"URL: {results['results'][0]['url']}")

    # Example 2: Code search
    print("\n💻 Example 2: Code Search")
    print("-" * 60)
    results = client.search_code("how to use asyncio in python")
    print(f"Found {len(results['results'])} code-related results")
    if results["results"]:
        print(f"First result: {results['results'][0]['title']}")
        print(f"URL: {results['results'][0]['url']}")

    # Example 3: Academic search
    print("\n🎓 Example 3: Academic Search")
    print("-" * 60)
    results = client.search_academic("transformer neural networks")
    print(f"Found {len(results['results'])} academic papers")
    if results["results"]:
        print(f"First paper: {results['results'][0]['title']}")
        print(f"URL: {results['results'][0]['url']}")

    # Example 4: Custom search with specific engines
    print("\n🎯 Example 4: Custom Search")
    print("-" * 60)
    results = client.search("docker kubernetes", engines=["github", "brave"])
    print(f"Found {len(results['results'])} results from GitHub and Brave")
    if results["results"]:
        print(f"First result: {results['results'][0]['title']}")
        print(f"Engine: {results['results'][0]['engine']}")

    print("\n✅ Quick start complete!")
    print("\n💡 Next steps:")
    print(
        "   - Import the client in your code: from searxng_client import SearXNGClient"
    )
    print("   - Check the README.md for detailed API documentation")
    print("   - Explore the available search engines and options")
    print("\n🌐 API Endpoint: https://searxng-production-b099.up.railway.app")


if __name__ == "__main__":
    main()
